### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: null safety migration for the package

### CODE REVIEW

**Review by dkrutskikh — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -45,13 +45,13 @@ jobs:
...
 
-  build-dart-2-9:
+  build-dart-2-12:
     name: Build on minimal supported dart version.
 
     runs-on: ubuntu-latest
 
     container:
-      image: google/dart:2.9.3
+      image: google/dart:2.12.0
```
[the remark this answers is not in the document]
Comment (incendial) [reply .github/workflows/build.yaml:54]: @dkrutskikh but is it correct? The name says` Build on **minimal** supported dart version.`,  so it looks like we should keep 2.12.0 or rename the pipeline.

File: pubspec.yaml [hunk 2]
```diff
@@ -6,29 +6,31 @@ repository: https://github.com/dart-code-checker/dart-code-metrics
...
 
 dev_dependencies:
-  coverage: ^0.14.0
-  http_multi_server: ^2.2.0       # for valid resolve dependencies while downgrade
-  mockito: ^4.1.1
-  pedantic: ^1.9.2
-  pub_semver: ^1.4.4
-  test: ^1.15.3
+  coverage: ^1.0.2
+  http_multi_server: ^3.0.0 # for valid resolve dependencies while downgrade
```
[the remark this answers is not in the document]
Comment (incendial) [reply pubspec.yaml:29]: Done

File: lib/src/utils/metric_utils.dart [hunk 3]
```diff
@@ -63,7 +63,7 @@ bool isReportLevel(MetricValueLevel level) =>
     level == MetricValueLevel.warning || level == MetricValueLevel.alarm;
 
 /// Returns user friendly string representations of [type].
-String userFriendlyType(Type type) {
+String userFriendlyType(Type? type) {
```
[the remark this answers is not in the document]
Comment (incendial) [reply lib/src/utils/metric_utils.dart:66]: Fixed

**Comment by GroovinChip** [comment]
Yay!!!

**Pull request merged**